storage_service/templates/snippets/packages_table.html

```diff
@@ -1,120 +1,22 @@
-          {% else %}
-          {% trans "None" %}
-          {% endif %}
-        </td>
-        <td>
-          {{ package.get_status_display }}
-          {% if package.status != 'DELETED' and package.status != 'FAIL'%}
-          (<a href="{% url 'package_update_status' package.uuid %}?next={{ request.path }}">{% trans "Update Status" %}</a>)
-          {% endif %}
-        </td>
-        <td>
-          {{ package.latest_fixity_check_datetime|default_if_none:"" }}
-        </td>
-        <td>
-          <a href="{% url 'package_fixity' package.uuid %}">{{ package.latest_fixity_check_result|yesno:_("Success,Failed,") }}</a>
-        </td>
-        <td>
-
-          {% if package.status != package.DELETED %}
-            <a href="{% url 'download_request' 'v2' 'file' package.uuid %}">{% trans "Download" %}</a>
-          {% endif %}
-
-          <!-- Replicas should not be re-ingestible -->
-          {% if package.package_type in 'AIP AIC' and not package.replicated_package %}
-            <a href="{% url 'aip_reingest' package.uuid %}?next={{ request.path }}">{% trans "Re-ingest" %}</a>
-          {% endif %}
-
-          {% if package.package_type in package.PACKAGE_TYPE_CAN_DELETE %}
-          <a href="#" class="request-delete"
-             data-package-type="{{ package.package_type }}"
-             data-package-uuid="{{ package.uuid }}"
-             data-package-pipeline="{{ package.origin_pipeline.uuid }}"
-             >{% trans "Request Deletion" %}</a>
-          {% endif %}
-
-          {% if package.package_type in package.PACKAGE_TYPE_CAN_DELETE_DIRECTLY and package.status != package.DELETED %}
-            <form method="POST" class="submit-confirm" action="{% url 'package_delete' package.uuid %}">
-              {% csrf_token %}
-              <button class="link" type="submit">{% trans "Delete" %}</button>
-              <div class="confirm-modal modal hide fade" tabindex="-1" role="dialog">
-                <div class="modal-header">
-                  <button type="button" class="close" data-dismiss="modal" aria-hidden="true">×</button>
-                  <h3>{% trans "Delete package" %}</h3>
-                </div>
-                <div class="modal-body">
-                  <p>
-                    {% blocktrans with type=package.package_type uuid=package.uuid %}
-                      Are you sure you want to delete this package ({{ type }}) with UUID <strong>{{ uuid }}</strong>?
-                    {% endblocktrans %}
-                  </p>
-                </div>
-                <div class="modal-footer">
-                  <button class="btn" data-dismiss="modal" aria-hidden="true">{% trans "Close" %}</button>
-                  <button class="btn btn-danger" type="submit">{% trans "Delete" %}</button>
-                </div>
-              </div>
-            </form>
-          {% endif %}
-
-        </td>
-      </tr>
+      {% include "snippets/package_row.html" %}
```

Comment:
It looks like on initial load we are rendering the full table, then immediately updating it via datatables — can we skip the initial render and display a "Loading..." row? Or, defer the datatable ajax request until the table needs updating (I couldn't find an easy option for that, but there may be one).

Reply:
Ah yes! I think it's [iDeferLoading](http://legacy.datatables.net/usage/options#iDeferLoading)

Reply:
`iDeferLoading` presented the problem that it needs the total number of records for the other navigations options to work even if the datatable hasn't been built. So as you suggested, instead I set a `Loading data from server` message and let the table to handle the initial rendering. I also verified the behavior in the Network tab of the devtools. Thank you for catching it!